rs/ic_os/os_tools/hostos_tool/src/main.rs

```diff
@@ -55,83 +55,82 @@ pub fn main() -> Result<()> {
             println!("{}", to_cidr(ipv6_address, network_info.ipv6_subnet));
             Ok(())
         }
-        Some(Commands::GenerateMacAddress { node_type }) => {
-            let config_map = config_map_from_path(Path::new(&opts.config))
-                .context("Please specify a valid config file with '--config'")?;
-            eprintln!("Using config: {:?}", config_map);
-
-            let network_info = NetworkInfo::from_config_map(&config_map)?;
-            eprintln!("Network info config: {:?}", &network_info);
-
-            let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file))
-                .context("Please specify a valid deployment file with '--deployment-file'")?;
-            eprintln!("Deployment config: {:?}", deployment_settings);
+        Some(Commands::GenerateMacAddress { ref node_type }) => {
+            let (_, mgmt_mac, deployment_settings) = get_config_settings(&opts)?;
 
             let node_type = node_type.parse::<NodeType>()?;
-            let mac = generate_mac_address(
-                &deployment_settings.deployment.name,
+            let generated_mac = generate_mac_address(
+                &mgmt_mac,
+                deployment_settings.deployment.name.as_str(),
                 &node_type,
-                deployment_settings.deployment.mgmt_mac.as_deref(),
             )?;
-            let mac = FormattedMacAddress::from(&mac);
-            println!("{}", mac.get());
+            let generated_mac = FormattedMacAddress::from(&generated_mac);
+            println!("{}", generated_mac.get());
             Ok(())
         }
         None => Err(anyhow!(
             "No subcommand specified. Run with '--help' for subcommands"
         )),
     }
 }
+
+fn get_config_settings(
+    opts: &HostOSArgs,
+) -> Result<(NetworkInfo, FormattedMacAddress, DeploymentSettings)> {
+    let config_map = config_map_from_path(Path::new(&opts.config)).context(format!(
+        "Failed to get config.ini settings for path: {}",
+        &opts.config
+    ))?;
+    eprintln!("Using config: {:?}", config_map);
+
+    let network_info = NetworkInfo::from_config_map(&config_map)?;
+    eprintln!("Network info config: {:?}", &network_info);
+
+    let deployment_settings =
+        get_deployment_settings(Path::new(&opts.deployment_file)).context(format!(
+            "Failed to get deployment settings for file: {}",
+            &opts.deployment_file
+        ))?;
+    eprintln!("Deployment config: {:?}", deployment_settings);
+
+    let mgmt_mac = match deployment_settings.deployment.mgmt_mac {
+        Some(ref config_mac) => {
+            let mgmt_mac = FormattedMacAddress::try_from(config_mac.as_str())?;
+            eprintln!(
+                "Using mgmt_mac address found in deployment.json: {}",
+                mgmt_mac.get()
+            );
+            mgmt_mac
+        }
+        None => get_ipmi_mac()?,
+    };
+
+    Ok((network_info, mgmt_mac, deployment_settings))
+}
```

Comment:
I created a helper function get_config_settings in setupos_tool and hostos_tool. wdyt? Cuts down on a lot of code repeat and makes the commands themselves much more readable.

Reply:
UPDATE: I removed the helper function because this code is going to eventually go away once we're on the new config (see: https://github.com/dfinity/ic/pull/1563), because all you'll have to do is grab the config object and that will have everything you need.